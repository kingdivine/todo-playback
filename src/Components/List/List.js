import React from "react";
import ListItem from "../ListItem/ListItem";
import { v4 as uuidv4 } from "uuid";

import "./List.scss";

const List = ({ todos, createTodo, deleteTodo, updateTodo }) => {
  const handleClickAdd = () => {
    const newTodo = {
      id: uuidv4(),
      name: uuidv4(),
      description: "Task description",
      createdDate: new Date(),
    };
    createTodo(newTodo);
  };
  return (
    <div className="list-container">
      <div>
        <button onClick={handleClickAdd}>Add Todo</button>
      </div>
      <div className="list-items-container">
        {todos.length > 0 ? (
          todos.map((todo) => (
            <ListItem
              key={todo.id}
              todo={todo}
              deleteTodo={deleteTodo}
              updateTodo={updateTodo}
            />
          ))
        ) : (
          <h3 className="empty-list-message"> No todos to display!</h3>
        )}
      </div>
    </div>
  );
};

export default List;
