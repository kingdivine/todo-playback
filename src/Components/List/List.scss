@import "../../styles/variables";

.list-items-container {
  width: 50%;
  margin: auto;
}

.empty-list-message {
  font-family: $primary-font;
}
