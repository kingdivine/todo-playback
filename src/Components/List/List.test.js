import React from "react";
import { render, cleanup } from "@testing-library/react";

import List from "./List";

afterEach(cleanup);

describe("<List/>", () => {
  const todos = [
    {
      id: "abc123",
      name: "Item 1",
      description: "This is the first item to complete",
      createdDate: new Date("Thu Apr 09 2020 16:36:07 GMT+0100 (BST)"),
    },
    {
      id: "def456",
      name: "Item 2",
      description: "This is the second item to complete",
      createdDate: new Date("Thu Apr 09 2020 16:37:07 GMT+0100 (BST)"),
    },
    {
      id: "ghi789",
      name: "Item 3",
      description: "This is the third item to complete",
      createdDate: new Date("Thu Apr 09 2020 16:38:07 GMT+0100 (BST)"),
    },
  ];
  test("renders empty list correctly", () => {
    expect(
      render(
        <List
          todos={[]}
          createTodo={jest.fn()}
          deleteTodo={jest.fn()}
          updateTodo={jest.fn()}
        />
      )
    ).not.toBeNull();
  });
  test("renders non-empty list correctly", () => {
    expect(
      render(
        <List
          todos={todos}
          createTodo={jest.fn()}
          deleteTodo={jest.fn()}
          updateTodo={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });
});
