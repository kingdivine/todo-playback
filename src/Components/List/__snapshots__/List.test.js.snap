// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List/> renders non-empty list correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="list-container"
      >
        <div
          class="list-items-container"
        >
          <div>
            <button
              class="add-btn"
            >
              Add Todo
            </button>
          </div>
          <div
            class="list-item"
          >
            <p
              class="item-name"
            >
              Item 1
            </p>
            <p
              class="item-description"
            >
              This is the first item to complete
            </p>
            <p
              class="item-date"
            >
              Yesterday at 4:36 PM
            </p>
            <button
              class="delete-btn"
            >
              Delete
            </button>
            <button
              class="edit-btn"
            >
              Edit
            </button>
          </div>
          <div
            class="list-item"
          >
            <p
              class="item-name"
            >
              Item 2
            </p>
            <p
              class="item-description"
            >
              This is the second item to complete
            </p>
            <p
              class="item-date"
            >
              Yesterday at 4:37 PM
            </p>
            <button
              class="delete-btn"
            >
              Delete
            </button>
            <button
              class="edit-btn"
            >
              Edit
            </button>
          </div>
          <div
            class="list-item"
          >
            <p
              class="item-name"
            >
              Item 3
            </p>
            <p
              class="item-description"
            >
              This is the third item to complete
            </p>
            <p
              class="item-date"
            >
              Yesterday at 4:38 PM
            </p>
            <button
              class="delete-btn"
            >
              Delete
            </button>
            <button
              class="edit-btn"
            >
              Edit
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="list-container"
    >
      <div
        class="list-items-container"
      >
        <div>
          <button
            class="add-btn"
          >
            Add Todo
          </button>
        </div>
        <div
          class="list-item"
        >
          <p
            class="item-name"
          >
            Item 1
          </p>
          <p
            class="item-description"
          >
            This is the first item to complete
          </p>
          <p
            class="item-date"
          >
            Yesterday at 4:36 PM
          </p>
          <button
            class="delete-btn"
          >
            Delete
          </button>
          <button
            class="edit-btn"
          >
            Edit
          </button>
        </div>
        <div
          class="list-item"
        >
          <p
            class="item-name"
          >
            Item 2
          </p>
          <p
            class="item-description"
          >
            This is the second item to complete
          </p>
          <p
            class="item-date"
          >
            Yesterday at 4:37 PM
          </p>
          <button
            class="delete-btn"
          >
            Delete
          </button>
          <button
            class="edit-btn"
          >
            Edit
          </button>
        </div>
        <div
          class="list-item"
        >
          <p
            class="item-name"
          >
            Item 3
          </p>
          <p
            class="item-description"
          >
            This is the third item to complete
          </p>
          <p
            class="item-date"
          >
            Yesterday at 4:38 PM
          </p>
          <button
            class="delete-btn"
          >
            Delete
          </button>
          <button
            class="edit-btn"
          >
            Edit
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
